python/tvm/topi/cuda/sparse.py

```diff
@@ -295,7 +295,14 @@ def is_valid_for_sparse_dense_padded(data, weight_data):
     """
     # pylint:disable=invalid-name
     warp_size = int(tvm.target.Target.current(allow_none=False).thread_warp_size)
-    m = get_const_tuple(data.checked_type.shape)[1]
+    # If there are multiple alter_ops in a model, the first alteration does not
+    # run type inference for the subsequent ones. In this case, we don't have
+    # the shape information, so we run the inferencer manually.
+    try:
```

Comment:
Is it possible to do it without try catch ?

Reply:
It doesn't seem to be possible without try catch. If I use `hasattr`, it throws an exception. And I don't know of any other way to check if the type has been inferred.

Reply:
Do you mean hasattr(data.checked_type, 'shape') throws error ?

Reply:
`hasattr(data, "checked_type")` throws

Reply:
I tried the logic with another member, it worked for me. Surprised to know u r getting exception :)

Reply:
@tkonolige : Would you please try as below:
 ```
    if hasattr(data, "checked_type"):
        m = get_const_tuple(data.checked_type.shape)[1]
    else:
        data_infered = relay.transform.InferType()(tvm.IRModule.from_expr(data))["main"]
        m = get_const_tuple(data_infered.ret_type.shape)[1]
```

Reply:
@ANSHUMAN87, yes, as I said, it throws an exception.

```
  File "tvm/_ffi/_cython/./packed_func.pxi", line 56, in tvm._ffi._cy3.core.tvm_callback
  File "/home/tristan/octoml/tvm/python/tvm/relay/op/nn/_nn.py", line 89, in alter_op_layout_sparse_dense
    return topi.nn.sparse_dense_alter_layout(attrs, inputs, tinfos, out_type)
  File "<decorator-gen-73>", line 2, in sparse_dense_alter_layout
  File "/home/tristan/octoml/tvm/python/tvm/target/generic_func.py", line 275, in dispatch_func
    return dispatch_dict[k](*args, **kwargs)
  File "/home/tristan/octoml/tvm/python/tvm/topi/cuda/sparse.py", line 408, in _alter_sparse_dense_layout
    and is_valid_for_sparse_dense_padded(inputs[0], inputs[1].data.asnumpy())
  File "/home/tristan/octoml/tvm/python/tvm/topi/cuda/sparse.py", line 301, in is_valid_for_sparse_dense_padded
    if hasattr(data, "checked_type"):
  File "/home/tristan/octoml/tvm/python/tvm/ir/expr.py", line 50, in checked_type
    raise ValueError("The type checker has not populated" " the checked_type for this node")
ValueError: The type checker has not populated the checked_type for this node
```